lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -625,9 +635,15 @@ def write(chunk)
                                             severity,
                                             ts_secs,
                                             ts_nanos)
-
           trace = record.delete(@trace_key)
-          entry.trace = trace if trace
+          if trace
+            if @enable_stackdriver_trace_autoformat &&
+               !STACKDRIVER_TRACE_REGEXP.match(trace)
```

Comment:
Just realized that a trace value that isn't in the right format but does have a slash will probably produce a log entry that will fail validation. If we adjust it unconditionally, then the resulting error might not make sense to the customers (e.g., "too many slashes" instead of "too few slashes"). We should confirm this behavior and, if my hunch is right, change this to a positive regexp match instead (i.e., only adjust if it has a certain form). WDYT?

Reply:
That's a valid concern. Agreed we should confirm the behavior.

Reply:
I found format of TraceID is documented and fully supported under [Trace API v1 (REST)](https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#Trace), and [Trace API (rpc) - google.devtools.cloudtrace](https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v1#google.devtools.cloudtrace.v1.Trace) - so I will rely on that to write the regexp matching.